services/api-server/src/simcore_service_api_server/api/routes/solvers.py

```diff
@@ -0,0 +1,170 @@
+from ...modules.catalog import CatalogApi
+from ..dependencies.application import get_reverse_url_mapper
+from ..dependencies.services import get_api_client
+from .solvers_faker import the_fake_impl
+
+NAMESPACE_SOLVER_KEY = uuidlib.UUID("ca7bdfc4-08e8-11eb-935a-ac9e17b76a71")
+
+# TODO: from urllib.request import pathname2url
+# TODO: from fastapi.responses import RedirectResponse
+
+logger = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@functools.lru_cache()
+def compose_solver_id(solver_key: SolverImageName, version: str) -> UUID:
+    # FIXME: this is a temporary solution. Should be image id
+
+    return uuidlib.uuid3(NAMESPACE_SOLVER_KEY, f"{solver_key}:{version}")
+
+
+## SOLVERS -----------------------------------------------------------------------------------------
+
+
+@router.get("", response_model=List[Solver])
+async def list_solvers(
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    """
+    Returns a list of the latest version of each solver
+    """
+    latest_solvers = [
+        Solver(
+            solver_url=url_for(
+                "get_solver_by_id",
+                solver_id=data["uuid"],
+            ),
+            **data,
+        )
+        for data in the_fake_impl.solvers
+    ]
+
+    return sorted(latest_solvers, key=attrgetter("name"))
+
+
+@router.get("/{solver_id}", response_model=Solver)
+async def get_solver_by_id(
+    solver_id: UUID,
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    try:
+        data = the_fake_impl.get(str(solver_id))
+        return Solver(
+            solver_url=url_for(
+                "get_solver_by_id",
+                solver_id=data["uuid"],
+            ),
+            **data,
+        )
+    except KeyError as err:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND) from err
+
+
+@router.get("/{solver_name:path}/{version}", response_model=Solver)
+async def get_solver_by_name_and_version(
+    solver_name: SolverImageName,
+    version: str,
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    try:
+        if version == LATEST_VERSION:
+            data = the_fake_impl.get_latest(solver_name)
+        else:
+            data = the_fake_impl.get_by_name_and_version(solver_name, version)
+
+    except KeyError as err:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND) from err
+
+    return Solver(
+        solver_url=url_for(
+            "get_solver_by_id",
+            solver_id=data["uuid"],
+        ),
+        **data,
+    )
+
+
+async def _list_solvers_impl(
+    catalog_client: CatalogApi = Depends(get_api_client(CatalogApi)),
+    url_for: Callable = Depends(get_reverse_url_mapper),
+):
+    # TODO: pagination
+    # TODO: deduce user
+    # TODO: filter and format as in docker listings ??
+
+    # FIXME: temporary hard-coded user!
+    user_id = 1
+
+    available_services = await catalog_client.get(
```

Comment:
consider using
```m = parse_obj_as(List[ServiceDockerData], available_services)```
maybe you can disable the validation since it is done already by the catalog. That might enable the usage of enums instead of strings for checking the type. finding that pattern would probably be also re-usable in other services.